src/app/api/utils/auth.ts

```diff
@@ -62,32 +62,29 @@ export const authOptions: AuthOptions = {
         },
       },
       token: AppConstants.OAUTH_TOKEN_URI,
+      // This function calls `profile` below, which converts the `FxaProfile` -> `Profile`.
       userinfo: {
         request: async (context) =>
-          fetchUserInfo(context.tokens.access_token ?? ""),
+          fetchUserInfo(
+            context.tokens.access_token ?? ""
+          ) as unknown as Promise<Profile>,
```

Comment:
@Vinnl does this clarify things or make it worse? :)

Reply:
For context, this is what this code did previous to any of this refactoring:

```
request: async (context) => {
  const response = await fetch(AppConstants.OAUTH_PROFILE_URI, {
    headers: {
      Authorization: `Bearer ${context.tokens.access_token ?? ""}`,
    },
  });
  const userInfo = (await response.json()) as Profile;
  return userInfo;
},
```

I'm actually not sure I totally understand why this worked before...

Reply:
I don't think this is right; if I log the return value of `fetchUserInfo`, it looks like this, which matches `FxaProfile`, not `Profile`:

```
{
    email: 'someaddress@mailinator.com',
    locale: 'en-GB,en-US;q=0.8,en;q=0.5,nl;q=0.3',
    amrValues: [ 'pwd', 'email' ],
    twoFactorAuthentication: false,
    metricsEnabled: true,
    uid: 'da4efbcfe51d466792e894accd119ff4',
    avatar: 'https://profile.stage.mozaws.net/v1/avatar/v',
    avatarDefault: false
  }
```

The type error does indicate that apparently `userInfo.request` does expect to return a `Profile`, so I suppose the return value is supposed to parse it into that as well?

I don't quite understand when `profile` is used then, but :shrug:

Reply:
Hmm OK I see now, the way I defined the `Profile` type initially appears to be wrong - that's supposed to be what I called `FxaProfile`, and then confusing, the return value of the `profile` callback is a `User` object, that I hadn't defined yet. Submitted https://github.com/mozilla/blurts-server/pull/3503 that I think fixes this - this branch is its target branch.